src/applications/pokemon_summary_screen/main.c

```diff
@@ -0,0 +1,2174 @@
+    summaryScreen->moveNameLoader = MessageLoader_Init(MESSAGE_LOADER_BANK_HANDLE, NARC_INDEX_MSGDATA__PL_MSG, message_bank_move_names, HEAP_ID_POKEMON_SUMMARY_SCREEN);
+    summaryScreen->playerName = Strbuf_Init(TRAINER_NAME_LEN + 1, HEAP_ID_POKEMON_SUMMARY_SCREEN);
+
+    if (summaryScreen->data->OTName != NULL) {
+        Strbuf_CopyChars(summaryScreen->playerName, summaryScreen->data->OTName);
+    }
+}
+
+static void FreeStrings(PokemonSummaryScreen *summaryScreen)
+{
+    MessageLoader_Free(summaryScreen->moveNameLoader);
+    MessageLoader_Free(summaryScreen->ribbonLoader);
+    MessageLoader_Free(summaryScreen->msgLoader);
+    sub_0200C560(summaryScreen->unk_684);
+    StringTemplate_Free(summaryScreen->strFormatter);
+    Strbuf_Free(summaryScreen->monData.speciesName);
+    Strbuf_Free(summaryScreen->monData.nickname);
+    Strbuf_Free(summaryScreen->monData.OTName);
+    Strbuf_Free(summaryScreen->strbuf);
+    Strbuf_Free(summaryScreen->playerName);
+}
+
+static int WaitSummaryScreenTransition(PokemonSummaryScreen *summaryScreen)
+{
+    if (IsScreenTransitionDone() == TRUE) {
+        PokemonSummaryScreen_LoadMonAnimation(summaryScreen);
+        PlayMonCry(summaryScreen);
+
+        if (summaryScreen->data->mode == PSS_MODE_SELECT_MOVE) {
+            return PSS_STATE_LEARN_MOVE;
+        } else if (summaryScreen->data->mode == PSS_MODE_CONDITION) {
+            return PSS_STATE_SETUP_POFFIN_FEED;
+        } else {
+            return PSS_STATE_HANDLE_INPUT;
+        }
+    }
+
+    return PSS_STATE_WAIT_TRANSITION;
+}
+
+static int HandleInput_Main(PokemonSummaryScreen *summaryScreen)
+{
+    if (summaryScreen->subscreenExit == TRUE) {
+        summaryScreen->data->returnMode = 1;
+        return PSS_STATE_TRANSITION_OUT;
+    }
+
+    if (gCoreSys.pressedKeysRepeatable & PAD_KEY_LEFT) {
+        ChangePage(summaryScreen, -1);
+        return PSS_STATE_HANDLE_INPUT;
+    }
+
+    if (gCoreSys.pressedKeysRepeatable & PAD_KEY_RIGHT) {
+        ChangePage(summaryScreen, 1);
+        return PSS_STATE_HANDLE_INPUT;
+    }
+
+    if (gCoreSys.pressedKeysRepeatable & PAD_KEY_UP) {
+        ChangeSummaryMon(summaryScreen, -1);
+        return PSS_STATE_HANDLE_INPUT;
+    }
+
+    if (gCoreSys.pressedKeysRepeatable & PAD_KEY_DOWN) {
+        ChangeSummaryMon(summaryScreen, 1);
+        return PSS_STATE_HANDLE_INPUT;
+    }
+
+    if (gCoreSys.pressedKeys & PAD_BUTTON_B) {
+        Sound_PlayEffect(SEQ_SE_DP_DECIDE);
+        summaryScreen->data->returnMode = 1;
+        return PSS_STATE_TRANSITION_OUT;
+    }
+
+    if (gCoreSys.pressedKeys & PAD_BUTTON_A) {
+        if (summaryScreen->data->mode == PSS_MODE_POFFIN && summaryScreen->page == PSS_MODE_CONDITION) {
+            Sound_PlayEffect(SEQ_SE_DP_DECIDE);
+            return TryFeedPoffin(summaryScreen);
+        }
+
+        if (summaryScreen->page == PSS_PAGE_BATTLE_MOVES) {
+            Sound_PlayEffect(SEQ_SE_DP_SYU01);
+            summaryScreen->subscreen = 0;
```

Comment:
What does this value do? Is it related to the subscreen mode constants defined in the header?

Reply:
I don't think it actually has anything to do with the actual subscreen, it seems like it's used mostly as a sort of state indicator for some of the information that pops up on top of the rest of the screen (like the move info panel). It definitely needs a rename.